### PR TITLE
chore: migrate source transform tests

### PR DESCRIPTION
**Pull request by sanpj2292**
## What are the changes introduced in this PR?

migrating source transformation tests to new format of component tests

## What is the related Linear task?

Resolves INT-2353

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3553/test-report.html

File: test/integrations/sources/segment/data.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
+import utils from '../../../../src/v0/util';
+
+const defaultMockFns = () => {
+  jest.spyOn(utils, 'generateUUID').mockReturnValue('97fcd7b2-cc24-47d7-b776-057b7b199513');
+};
+
+export const data = [
```
Comment (lvrach) [inline test/integrations/sources/segment/data.ts:7]: Should we add type checking here as well ?

```suggestion
export const data : TestCaseData[] = [
```

File: test/integrations/destinations/pagerduty/router/data.ts [hunk 2]
```diff
@@ -267,5 +267,11 @@ export const data = [
         },
       },
     },
+    mockFns: (_) => {
+      jest.spyOn(Date, 'now').mockImplementation(() => {
+        console.log('Pagerduty router mock');
```
Comment (lvrach) [inline test/integrations/destinations/pagerduty/router/data.ts:272]: Is the log helpful here?  Why are not using it in other test cases 

```suggestion
```

File: test/integrations/sources/appcenter/data.ts [hunk 3]
```diff
@@ -0,0 +1,372 @@
+import utils from '../../../../src/v0/util';
+
+export const data = [
+  {
+    name: 'appcenter',
+    description: 'test-0',
+    module: 'source',
+    version: 'v0',
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/sources/appcenter/data.ts:8]: In source transformation we have `v0` and `v1` versions. This version indicates we are using `v0` version of source transformation

File: test/integrations/destinations/pagerduty/router/data.ts (hunk 2 above)
Comment (sanpj2292) [reply test/integrations/destinations/pagerduty/router/data.ts]: Done

File: test/integrations/sources/appcenter/data.ts [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+          {
+            text: 'Hello from your abc-test app in App Center!',
+            sent_at: '2023-01-02T07: 53: 28.3117824Z',
+            url: 'https://appcenter.ms/users/abc-rudderstack.com/apps/abc-test',
+          },
+        ],
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+      },
+      pathSuffix: '',
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/sources/appcenter/data.ts:21]: Not sure about including this for all. For now, we can let this be as is and if needed look into some kind of optimisation at a later point of time

[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/sources/appcenter/data.ts:21]: Not sure about including this for all. For now, we can let this be as is and if needed look into some kind of optimisation at a later point of time

**Review by koladilip — APPROVED** [review]

**Review by lvrach — APPROVED** [review]

**Comment by sanpj2292** [comment]
> How are we handling shopify source test cases ? I don't see those in the new structure

Missed migrating it. Added the migrated tests now.

File: test/integrations/sources/appcenter/data.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/sources/appcenter/data.ts:8]: Resolving this conversation.

File: go/webhook/testcases/testdata/testcases/auth0/empty_batch.json [hunk 5]
```diff
@@ -11,11 +11,10 @@
   },
   "output": {
     "response": {
-      "status": 200,
-      "body": "OK"
+      "status": 400,
```
Comment (krishna2020) [inline go/webhook/testcases/testdata/testcases/auth0/empty_batch.json:14]: why change in status ? Is there any corresponding change done for this ?

Comment (sanpj2292) [reply go/webhook/testcases/testdata/testcases/auth0/empty_batch.json:14]: The integration test was failing to validate status. Hence had to make this change.
> Is there any corresponding change done for this ?

On server side no changes are done.

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**